specification/offazurespringboot/resource-manager/Microsoft.OffAzureSpringBoot/preview/2023-01-01-preview/springbootdiscovery.json

```diff
@@ -1319,6 +1431,13 @@
             "name": "ProvisioningState",
             "modelAsString": true
           }
+        },
+        "tags": {
```

Comment:
This resource is inherited from proxyResource, and for now arm don't support tags for proxyResouces, but according to our requirements we had to implements it in the nested properties.

Reply:
At the time I wrote the feedback it was a tracked resource. For a proxy resource the feedback is that it's best not to recreate the tags functionality if it can be avoided.

Tags V2 will definitely support proxy resources (I don't know the status of the new functionality, but Jin Soon Lim should be able to give you the latest). Your RPs tags won't behave the same as ARM tags and won't integrate, won't be searchable in ARG, won't be usable with Azure Policy, etc.

If you still decide you absolutely MUST implement your own tags on a proxy resource, you need to choose a different name, e.g. resourceTags (or something else depending on what these tags are for) to distinguish them from ARM tags.

Reply:
Hi @mentat9 ,

We are aware that ARM team is implementing tags for proxy resources. 

We engaged with Akif Heren, Rich Thorne, Evan Hissey and others.

We got the go ahead to use the property name "tags" inside the Properties section, as the Properties section is owned by RP. 

We need to do this for some deliverables that are scheduled before the TagsV2 are implemented for Proxy resources.

Reply:
OK, given your discussion with ARM team members, including a property with "tags-like" behavior in the property bag is OK.

But you need to rename it to something else (e.g. labels, targetTags, something different/better). You aren't providing ARM tags functionality (either V1 or V2) and the "tags" name is reserved for ARM anyway. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope).

Also, when you add support for tags V2 and have both a "tags" and "properties.tags" property it would produce an even more confusing API.

@cemheren what do you think about the naming requirement?

Reply:
Thanks for adding me, I'm fully supportive of whatever API review board requirements we have. Keeping a consistent and clear API acrross Azure should always be the top priority. 

From naming perspective I don't have strong opinions, something like "labels" do make sense to me. @vegajula We should do whatever makes it easier here to migrate to top level tags once we can start supporting them. I stared seeking a list of proxy types from the migrate team for us to start targeting as the first group of proxy types to onboard to centralized tags.